spirv_glsl.cpp

```diff
@@ -2437,22 +2437,32 @@ string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtyp
 		break;
 	}
 
+	auto *lod_constant = maybe_get<SPIRConstant>(lod);
+	bool zero_lod = lod_constant && lod_constant->scalar_f32() == 0.0f;
+
 	if (op == "textureLod" || op == "textureProjLod")
 	{
-		if (is_legacy_es())
+		if (is_legacy_es() && !zero_lod)
```

Comment:
Probably cleaner to just nest the ifs to avoid the weird else if, if (is_legacy_es()) { if (!zero_lod) } ...